### DISCUSSION TITLE
Correct place for email validation?

### DISCUSSION THREAD

**Discussion by markhuge** [ory/kratos#3122] [Q&A] [answered]
It seems like the right place to validate email address strings (i.e. `foo@bar` or `foo` being invalid and `foo@bar.com` being valid) would be in the [before-registration hook](https://www.ory.sh/docs/kratos/hooks/configure-hooks#action-triggers).

Is this the case, or is there a config elsewhere I should be using instead?

**Comment by Benehiko** [comment, marked as answer]
Hi @markhuge 

Yes, you can use a webhook for this to do a check against a "block list" or want control over what you deem as a valid email address. 

Another approach is to have the Identity Schema do the validation according to some pattern
https://www.ory.sh/docs/kratos/manage-identities/best-practices#sanitize-usernamestraits. 

Also see https://github.com/ory/kratos/discussions/1617

**Reply by markhuge** [reply]
Thanks!
